lib/mongo/srv/resolver.rb

```diff
@@ -51,13 +61,18 @@ def timeout
         options[:timeout] || Monitor::DEFAULT_TIMEOUT
       end
 
-      # Obtains all of the SRV records for a given hostname.
+      # Obtains all of the SRV records for a given hostname. If a srv_max_hosts
+      # is specified and it is greater than 0, return maximum srv_max_hosts records.
       #
       # In the event that a record with a mismatched domain is found or no
       # records are found, if the :raise_on_invalid option is true,
       # an exception will be raised, otherwise a warning will be logged.
       #
       # @param [ String ] hostname The hostname whose records should be obtained.
+      # @param [ String | nil ] srv_service_name The SRV service name for the DNS query.
+      #   If nil, 'mongodb' is used.
+      # @param [ Integer | nill ] srv_max_hosts The maximum number of records to return.
+      #   If this value is zero or nil, return all of the records.
```

Comment:
I think it would be better to change zero to nil sometime during client construction and everywhere downstream of the client require and expect (with validation, if necessary) that the value is either nil or a positive integer.